test/parallel/test_mxnet2.py

```diff
@@ -0,0 +1,1546 @@
+# Copyright 2018 Uber Technologies, Inc. All Rights Reserved.
+# Modifications copyright (C) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import os
+import sys
+import itertools
+import unittest
+from distutils.version import LooseVersion
+
+import pytest
+import numpy as np
+
+sys.path.append(os.path.join(os.path.dirname(__file__), os.pardir, 'utils'))
+
+from common import skip_or_fail_gpu_test
+
+try:
+    import mxnet as mx
+    from mxnet.base import MXNetError
+    from mxnet.test_utils import almost_equal, same
+    import horovod.mxnet as hvd
+
+    has_gpu = mx.context.num_gpus() > 0
+
+    ccl_supported_types = set(['int32', 'int64', 'float32', 'float64'])
+
+    HAS_MXNET = True
+except ImportError:
+    has_gpu = False
+    HAS_MXNET = False
+
+# Set environment variable to enable adding/removing process sets after initializing Horovod.
+os.environ["HOROVOD_DYNAMIC_PROCESS_SETS"] = "1"
+
+
+@pytest.mark.skipif(not HAS_MXNET, reason='MXNet unavailable')
+@pytest.mark.skipif(LooseVersion(mx.__version__) < LooseVersion('2.0.0'), reason='MXNet v2 tests')
+class MXTests(unittest.TestCase):
```

Comment:
Can we call this `MX2Tests` and derive it from `MXTests` so we don't copy all that code and have to maintain it in two places? Then `MX2Tests` can override and reimplement those tests that don't work in MXNet 2.0 only.

Will be interesting how the `@pytest.mark.skipif` handles the class hierarchy.